src/main/java/tc/oc/pgm/api/match/Match.java

```diff
@@ -393,4 +393,6 @@ default boolean finish() {
    * @return If the {@link Match} was just ended.
    */
   boolean calculateVictory();
+
+  Listener getListener(Class<? extends Listener> listenerClass);
```

Comment:
This doesn't seem like the right solution to this problem. Doesn't `CombatLogTracker` cancel the leave event or something?

Reply:
It does, but the event is fired after `JoinMatchModule#leave(MatchPlayer)`.

Reply:
Code will need to be redesigned so that the event is triggered first, and if not cancelled, then the message is shown.

Doing this getListener logic is a complete hack

Reply:
Should I close this PR and submit a new one for that or just add it to this one?

Reply:
You can just ammend the commit

Reply:
Yea, changing the event cancellation logic is the right way to go.